server/tests/use_cases/test_count_volunteers.py

```diff
@@ -0,0 +1,358 @@
+            "end_time": 5
+        }
+    ]
+
+    request = Object(filters={"start_after":1, "end_before":6})
+    expected = [{"start_time":1, "end_time":2, "count":1},
+                {"start_time":2, "end_time":3, "count":2},
+                {"start_time":3, "end_time":4, "count":3},
+                {"start_time":4, "end_time":5, "count":2},
+                {"start_time":5, "end_time":6, "count":1}]
+
+    compare(expected, shifts_data, request)
+
+# X...Y...Z...Y...Z...X
+def test_many_overlapping3_shifts():
+    shifts_data = [
+        {
+            "code": "3",
+            "worker": "volunteer@slu.edu",
+            "shelter": 1,
+            "start_time": 2,
+            "end_time": 4
+        },
+        {
+            "code": "1",
+            "worker": "volunteer@slu.edu",
+            "shelter": 1,
+            "start_time": 1,
+            "end_time": 6
+        },
+        {
+            "code": "2",
+            "worker": "volunteer@slu.edu",
+            "shelter": 1,
+            "start_time": 3,
+            "end_time": 5
+        }
+    ]
+
+    request = Object(filters={"start_after":1, "end_before":6})
+    expected = [{"start_time":1, "end_time":2, "count":1},
+                {"start_time":2, "end_time":3, "count":2},
+                {"start_time":3, "end_time":4, "count":3},
+                {"start_time":4, "end_time":5, "count":2},
+                {"start_time":5, "end_time":6, "count":1}]
+
+    compare(expected, shifts_data, request)
+
+# X...Y...Z...X...Z...Y
+def test_many_overlapping4_shifts():
+    shifts_data = [
+        {
+            "code": "3",
+            "worker": "volunteer@slu.edu",
+            "shelter": 1,
+            "start_time": 3,
+            "end_time": 5
+        },
+        {
+            "code": "1",
+            "worker": "volunteer@slu.edu",
+            "shelter": 1,
+            "start_time": 1,
+            "end_time": 4
+        },
+        {
+            "code": "2",
+            "worker": "volunteer@slu.edu",
+            "shelter": 1,
+            "start_time": 2,
+            "end_time": 6
+        }
+    ]
+
+    request = Object(filters={"start_after":1, "end_before":6})
+    expected = [{"start_time":1, "end_time":2, "count":1},
+                {"start_time":2, "end_time":3, "count":2},
+                {"start_time":3, "end_time":4, "count":3},
+                {"start_time":4, "end_time":5, "count":2},
+                {"start_time":5, "end_time":6, "count":1}]
+
+    compare(expected, shifts_data, request)
+
+def test_workers_in_range():
+    shifts_data = [
+       {
+            "code": "1",
+            "worker": "volunteer@slu.edu",
+            "shelter": 1,
+            "start_time": 1,
+            "end_time": 6
+        }
+    ]
+
+    request = Object(filters={"start_after":1, "end_before":4})
+    expected = [{"start_time":1, "end_time":4, "count":1}]
+
+    compare(expected, shifts_data, request)
+
+def test_workers_in_range_overlap():
```

Comment:
Could you please explain this to me? I am not understanding how duplicates are handled.

Reply:
From our side discussion: there won't be duplicates in the database (same worker cannot be signed up for two overlapping shifts at the same time). I updated the test cases to make it clear that these are different workers that result in overlaps.